js/scripts.js

```diff
@@ -18,47 +13,55 @@ const addLi = (title, author) => {
   return li;
 };
 
-const refresh = () => {
-  bookList.innerHTML = '';
-  const storedBooks = JSON.parse(localStorage.books);
-  storedBooks.forEach(
+const popLi = () => {
+  books.forEach(
     (book) => {
       bookList.appendChild(addLi(book.title, book.author));
     },
   );
 };
 
-const addBook = () => {
-  const title = document.getElementById('enterTitle').value;
-  const author = document.getElementById('enterAuthor').value;
-  const storedBooks = JSON.parse(localStorage.books);
-  storedBooks.push(new Book(title, author));
-  localStorage.setItem('books', JSON.stringify(storedBooks));
-  document.getElementById('enterTitle').value = '';
-  document.getElementById('enterAuthor').value = '';
-  refresh();
+const refresh = () => {
+  bookList.innerHTML = '';
+  localStorage.setItem('books', JSON.stringify(books));
+  popLi();
 };
 
-const removeBook = (e) => {
-  const tgt = e.target;
-  if (tgt.tagName === 'BUTTON') {
-    const title = e.path[1].firstChild.data.split(' - ')[0];
-    let storedBooks = JSON.parse(localStorage.books);
-    storedBooks = storedBooks.filter((el) => el.title !== title);
-    localStorage.setItem('books', JSON.stringify(storedBooks));
-    tgt.closest('li').remove();
+class Book {
+  constructor(title, author) {
+    this.title = title;
+    this.author = author;
   }
-  refresh();
-};
 
-bookList.addEventListener('click', removeBook);
-document.getElementById('btnAdd').addEventListener('click', addBook);
+  addBook() {
+    this.title = document.getElementById('enterTitle').value;
+    this.author = document.getElementById('enterAuthor').value;
+    books.push({ title: this.title, author: this.author });
+    document.getElementById('enterTitle').value = '';
+    document.getElementById('enterAuthor').value = '';
+    refresh();
+  }
+
+  removeBook(e) {
+    const tgt = e.target;
+    if (tgt.tagName === 'BUTTON') {
+      [this.title] = e.composedPath()[1].firstChild.data.split(' - ');
+      books = books.filter((el) => el.title !== this.title);
+      tgt.closest('li').remove();
+    }
+    refresh();
+  }
+}
```

Comment:
- [ ] We are failing in one of the requirements as concerns deleting a book. Look at the screenshots below for reference.

**Firstly, a screenshot from the previous task, for emphasis**
| --- |
| ![earlier_requirement](https://user-images.githubusercontent.com/45002654/133308130-20ef6361-d196-4352-a178-f7ea892ac571.PNG) |

**Secondly, the specific requirement in this task that requires us to maintain the functionality**
| --- |
| ![new_requirement](https://user-images.githubusercontent.com/45002654/133308440-9410e95a-1f00-41b5-9aff-bca040c72391.PNG) |

## The problem

Right now, the app allows the addition of duplicate books(**_books with the same title_**), and when the user tries to remove just one, all the books with the same title will all be deleted. We are required to delete only one book at a time.

## Remedy

Let the app only accept unique books. OR, add a unique ID field to a Book object, so we may filter the books array with id property which is unique and not the title property.

Reply:
https://github.com/lfmnovaes/awesomebooks/blob/8f451839572a931aa62d3229c54ec15a14993c34/js/scripts.js#L5
We chose to create a unique ID for each book created and deleting using it.